### ISSUE TITLE
MD018 false positive inside code blocks using hl_lines

### ISSUE THREAD

**Issue by jacobrosenthal** [markdownlint/markdownlint#176]
Dont quite understand this yet, but this example false positives

https://gist.github.com/jacobrosenthal/e49f7d225b40b2bcc59a4f5d029776bc

```
test.md:5: MD018 No space after hash on atx style header
test.md:5: MD025 Multiple top level headers in the same document
```

**Comment by markeganfuller** [comment]
This seems to be an issue with the underlying parser Kramdown, specifically [this regex](https://github.com/gettalong/kramdown/blob/master/lib/kramdown/parser/gfm.rb#L129)

Which will match:

````
```example
one
```
````

but not

````
```example fail
two
```
````

**Comment by HariSekhon** [comment]
I'm hitting this issue in hash comments inside any code blocks for shell and python like this:

```shell
# comment
somecommand
```

```python
# comment
somecode
```

Where the comments are at the top like this I've moved them out of the code blocks so there is no precedeing `#` but where code blocks contain useful comments mid way through this is not practical:

```shell
command1 &&
command2
# this one is really necessary to work around github issue xxx...
command3 ||
command4
```

eg:
```
celery.md:153: MD022 Headers should be surrounded by blank lines
celery.md:158: MD022 Headers should be surrounded by blank lines
celery.md:153: MD025 Multiple top level headers in the same document
celery.md:158: MD025 Multiple top level headers in the same document
```

I don't want to disable `MD022` and `MD025` entirely in [.mdlrc](https://github.com/HariSekhon/Knowledge-Base/blob/main/.mdlrc) because I'm already excluding more and more and these are two useful checks that find out when I've ported my private KB in conf format to md format but left single hash comments which are not the right level of heading.

**Comment by jaymzh** [comment]
Yeah this is the fact that the kramdown parser doesn't see that as a codeblock. But we already override this stuff, so this should be a reasonably easy fix.
